### ISSUE TITLE
CPS transform breaks logical short-circuiting

### ISSUE THREAD

**Issue by dritchie** [probmods/webppl#58]
Code such as 
```
if (x && predfn()) { dosomething(); }
```
gets transformed to something like
```
var _s1 = x;
var _s2 = predfn();
var _test1 = _s1 && _s2;
if (_test1) { dosomething(); }
```
which is obviously problematic when `predfn` shouldn't be called unless `x` is true.

**Comment by ngoodman** [comment]
yep, this is a bug. (we used to not support short-circuiting ops for this reason... they must have gotten added at some point in such a way that && is treated as &, and so on.)

should treat these ops as control operators. a simple solution is for `a&&b` to become `a?true:b`.

**Comment by stuhlmueller** [comment]
Fixed by @kimmyg in 2a543ed52147758472d172641727526fa0f3c821.
